File: Recursion/Practice recursion/4_Linear Search using Recursion/Solution.py
# cook your dish here
def linear_search(Arr, n, x):
    #Write your code here
    if( len(Arr) == n ):
        return -1
        
    if( Arr[n] == x):
        return n
        
    return linear_search(Arr, n+1 , x )

n, x = map(int, input().split())
Arr = list(map(int, input().split()))
print(linear_search(Arr, 0, x))

File: Recursion/Practice recursion/4_Linear Search using Recursion/Solution.c
#include <stdio.h>
#include <stdlib.h>

int linearSearch(int arr[], int size, int x) {
    //Write your code here
    if(size==0)
    {
        return -1;
    }
    
    if( arr[size-1]== x )
    {
        return size-1;
    }
    
    
    return linearSearch(arr, size-1 , x);
}

int main() {
    int n, x;
    scanf("%d %d", &n, &x);
    int *arr = (int*)malloc(n * sizeof(int));
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }
    printf("%d\n", linearSearch(arr, n, x));
    free(arr);
    return 0;
}
